apis/k8ssandra/v1alpha1/k8ssandracluster_webhook_test.go

```diff
@@ -319,6 +320,114 @@ func testNumTokens(t *testing.T) {
 
 }
 
+func testRackNameValidation(t *testing.T) {
+	required := require.New(t)
+
+	createNamespace(required, "racknames-namespace")
+
+	createCluster := func(topLevelRacks, dcLevelRacks []string) *K8ssandraCluster {
+		cluster := createMinimalClusterObj("racknames-test", "racknames-namespace")
+		for _, rackName := range topLevelRacks {
+			cluster.Spec.Cassandra.Racks = append(
+				cluster.Spec.Cassandra.Racks, v1beta1.Rack{Name: rackName})
+		}
+		cluster.Spec.Cassandra.Datacenters[0].Meta = EmbeddedObjectMeta{Name: "dc1"}
+		for _, rackName := range dcLevelRacks {
+			cluster.Spec.Cassandra.Datacenters[0].Racks = append(
+				cluster.Spec.Cassandra.Datacenters[0].Racks, v1beta1.Rack{Name: rackName})
+		}
+		return cluster
+	}
+
+	var oldCluster, newCluster *K8ssandraCluster
+
+	// DC-level declaration
+	oldCluster = createCluster(nil, []string{"rack1", "rack2"})
```

Comment:
These tests might be a good candidate for a table driven style of testing https://dave.cheney.net/2019/05/07/prefer-table-driven-tests